Extension/src/LanguageServer/extension.ts

```diff
@@ -1162,3 +1162,39 @@ export function UpdateInsidersAccess(): void {
         void vscode.commands.executeCommand("workbench.extensions.installExtension", "ms-vscode.cpptools", { installPreReleaseVersion: true }).then(undefined, logAndReturn.undefined);
     }
 }
+
+export async function PreReleaseCheck(): Promise<void> {
+    // First we need to make sure the user isn't already on a pre-release version.
+    if (!util.packageJson.__metadata?.preRelease === true) {
+        // Get the info on the latest version from the marketplace to check if there is a pre-release version available.
+        const response = await fetch('https://marketplace.visualstudio.com/_apis/public/gallery/extensionquery', {
+            method: 'POST',
+            headers: {
+                Accept : 'application/json; api-version=3.0-preview',
+                'Content-Type' : 'application/json',
+            }, 
+            body: '{"filters": [{"criteria": [{"filterType": 7, "value": "ms-vscode.cmake-tools"}]}], "flags": 529}'
```

Comment:
Is this supposed to be ms-vscode.cpptools instead of cmake-tools?

Reply:
Yep, I copied the example a little too literally.